python/audit/cef_logger.py

```diff
@@ -0,0 +1,77 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+from collections import OrderedDict
+from typing import Tuple, Union
+
+from django.conf import settings
+from django.utils import timezone as tz
+
+from audit.apps import AuditConfig
+from audit.models import AuditLog, AuditLogOperationResult, AuditSession
+
+
+audit_log = logging.getLogger(AuditConfig.name)
+
+
+class CEFLogConstants:
+    cef_version: str = "CEF: 0"
+    device_vendor: str = "Arenadata Software"
+    device_product: str = "Arenadata Cluster Manager"
+    adcm_version: str = settings.ADCM_VERSION
+    operation_name_session: str = "User logged"
+    extension_keys: Tuple[str] = ("actor", "act", "operation", "resource", "result", "timestamp")
+
+
+def cef_log(
+    audit_instance: Union[AuditLog, AuditSession],
+    signature_id: str,
+    severity: int = 1,
+    empty_resource: bool = False,
+) -> None:
+    extension = OrderedDict.fromkeys(CEFLogConstants.extension_keys, None)
+    extension["timestamp"] = str(tz.now())
+
+    if isinstance(audit_instance, AuditSession):
+        operation_name = CEFLogConstants.operation_name_session
+        if audit_instance.user:
```

Comment:
Field `user` for `AuditSession` model could be `User` or `None`. Please add an explicit check for `None`

Reply:
done